blue_ai/envs/custom_decay.py

```diff
@@ -0,0 +1,26 @@
+import torch.nn as nn
+import torch
+from torch.nn.modules import MSELoss, Sequential
+
+
+class PositivePenaltyLoss(nn.Module):
+    known = MSELoss()
+
+    """
+    Uses a combinations of MSE Loss with a RELU penalty, meaning that positive 
+    weights will always have a large penalty 
+    """
+
+    def __init__(self, alpha=0.2):
+        super(PositivePenaltyLoss, self).__init__()
+        self.policy_hook: Sequential = None
+        self.alpha = alpha
+
+    def forward(self, inputs: torch.Tensor, targets: torch.Tensor):
+        net = next(self.policy_hook.parameters())
+        # This is actually a ReLU function
+        # https://discuss.pytorch.org/t/why-are-there-3-relu-functions-or-maybe-even-more/5891/2
+        positive_penalty = torch.clamp(net, min=0).sum()
```

Comment:
The positive weights should be squared before taking the mean to get the decay-proportional-to-size effect.
Use mean instead of sum to make the meaning of your new penalty consistent across networks of different sizes.

Reply:
I messed around with this for quite a while. We just don't get the same results without using a sum. We would need to apply a coefficient of 1000, but that kind of defeats the point of using a mean. Let's stick with the ReLU activation method.